Azure/sonic-sairedis
# For sflow feature, SAI changes needed for sonic-vs platform

- _What I did_

Add SAI library changes to support sflow in sonic-vs.

- _How I did it_

When sflow is configured using CLI or other supported mechanisms, the orch-agent invokes 
the SAI API set_port_attribute(). For the sflow feature, his API is used to set the SAI_PORT_ATTR_INGRESS_SAMPLEPACKET_ENABLE attribute of a netdev port. The value stored for this attribute is a samplepacket object, which essentially contains various sampling attributes (sampling rate, sampler group etc.) associated with the port.

When sampling is disabled on a port, the  SAI_PORT_ATTR_INGRESS_SAMPLEPACKET_ENABLE attribute  is set to a null object. 
When sampling is enabled on a port, the  SAI_PORT_ATTR_INGRESS_SAMPLEPACKET_ENABLE attribute is set to a valid samplepacket object. 

For sonic-vs, in the absence of a real ASIC programming, the 'tc' command is used instead to configure the sampling parameters inside the kernel. In this PR we invoke the appropriate 'tc' command, based on the requested config actions (disable or enable or update). 


- _How to verify it_

  - Using https://github.com/Azure/sonic-buildimage/pull/3251 changes, build a sonic image
  - Using https://github.com/Azure/sonic-utilities/pull/592 changes, use the CLIs to configure sflow globally and on a port. Thereafter, use the show CLI to verify the changes made. 
  - Use the logs to verify the invocation of the proposed change in this  PR.

- _Description for the changelog_

   - Update: Added vslib changes to support sflow in sonic-vs
 
- _Additional Information_
  Please **DO NOT MERGE** until these PRs are merged:
  https://github.com/Azure/sonic-buildimage/pull/3251

Signed-off-by: Rakesh Datta <rakesh.datta@dell.com>
